src/frontend/web_application/src/components/form/InputText/index.jsx

```diff
@@ -1,35 +1,52 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
 import './style.scss';
 
-const InputText = ({ expanded, theme, bottomSpace, className, hasError, ...props }) => {
-  const inputTextClassName = classnames(
-    'm-input-text',
-    {
-      'm-input-text--expanded': expanded,
-      'm-input-text--light': theme === 'light',
-      'm-input-text--bottom-space': bottomSpace,
-      'm-input-text--error': hasError,
-    },
-    className
-  );
+class InputText extends PureComponent {
+  static propTypes = {
+    expanded: PropTypes.bool,
+    theme: PropTypes.string,
+    bottomSpace: PropTypes.bool,
+    hasError: PropTypes.bool,
+    className: PropTypes.string,
+    inputRef: PropTypes.func,
```

Comment:
I'm not sure about this, by default input ref property gets a null instead of a function. It has more sense to set undefined as default value (like when it is not specified if I'm correct): `<input ref={undefined} /> === <input />`.

And well actually we do the same using `props` that have been transferred to input:

```jsx
<InputText name="foo" ref={inputRef => this.inputFoo = inputRef} />
```

 but it is not well named. I saw a more declarative syntax that should be better:

```jsx
<InputText inputProps={{ name: 'foo', ref: inputRef => this.inputFoo = inputRef }} />
```

What do you think ?

(in order to not have breaking changes or massive refactor, we should keep the actual props transmission: `<input {...inputProps} {...props} />`)

Reply:
ok about `inputRef` default to `undefined`.
I don't get what you mean about the `ref` prop, is it only related to syntax? adding a `inputRef` prop is the right way to pass the `ref` prop to child component according to docs.